### PR TITLE
MLPAB-1582 Sort out sending of reduced-fee-requested

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: internal/app/document_store.go [hunk 1]
```diff
@@ -79,50 +79,23 @@ func (s *documentStore) UpdateScanResults(ctx context.Context, lpaID, s3ObjectKe
...
-			s3Keys = append(s3Keys, d.Key)
+			dynamoKeys = append(dynamoKeys, dynamo.Key{PK: d.PK, SK: d.SK})
 		}
 	}
 
 	if len(dynamoKeys) == 0 {
 		return nil
 	}
 
-	if err := s.s3Client.DeleteObjects(ctx, s3Keys); err != nil {
```
Comment (hawx) [inline internal/app/document_store.go:122]: not needed now https://github.com/ministryofjustice/opg-modernising-lpa/pull/822

**Review by andrewpearce-digital — APPROVED** [review]

File: internal/page/donor/upload_evidence.go [hunk 2]
```diff
@@ -118,6 +118,16 @@ func UploadEvidence(tmpl template.Template, payer Payer, documentStore DocumentS
...
+					if len(documents.NotScanned()) > 0 {
+						logger.Print("attempt to pay with unscanned documents on lpa:", lpa.UID)
+						data.Errors = validation.With("upload", validation.CustomError{Label: "errorGenericUploadProblem"})
+						return tmpl(w, data)
+					}
+
+					if err := documentStore.Submit(r.Context(), lpa, documents); err != nil {
+						return err
+					}
+
```
Comment (acsauk) [inline internal/page/donor/upload_evidence.go:130]: Have we thought about how we handle people not paying or being unable to pay? Might be worth talking through with vega to see how they are expecting case workers to work evidence provided cases where payments have not been successful.

Comment (hawx) [reply internal/page/donor/upload_evidence.go:130]: It's a thing for us too since we only have the happy path for payment confirmation

Comment (acsauk) [reply internal/page/donor/upload_evidence.go:130]: Cool lets leave as is for now and worry about this in a future PR 👍

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**